pandas/core/arrays/datetimes.py

```diff
@@ -134,9 +134,11 @@ def f(self):
             return result
 
         if field in self._object_ops:
-            result = fields.get_date_name_field(values, field)
-            result = self._maybe_mask_results(result, fill_value=None)
-
+            if field == "isoformat":
+                result = fields.get_datetime_isoformats(values)
+            else:
+                result = fields.get_date_name_field(values, field)
+                result = self._maybe_mask_results(result, fill_value=None)
```

Comment:
is maybe_mask_results necessary for the isoformat case?  i.e. does it handle NaT?